server/controllers/boards.js

```diff
@@ -0,0 +1,158 @@
+const asyncHandler = require("express-async-handler");
+const Board = require('../models/Board');
+const Card = require('../models/Card');
+const Column = require('../models/Column');
+
+exports.createBoard = asyncHandler(async (req, res) => {
+  const { title } = req.body;
+  if (title) {
+    try {
+      const inProgress = new Column({ title: 'In Progress' });
+      const completed = new Column({ title : 'Completed' });
+      
+      await inProgress.save();
+      await completed.save();
+
+      const board = new Board({ title, columns: [inProgress._id, completed._id]});
+      const doc = await board.save();
+
+      res.status(201).json({
+        success: {
+          boardId: doc._id,
+        }
+      });
+    } catch (err) {
+      res.status(500);
+      console.error(err);
+    }
+  } else {
+    res.status(400).send({
+      message: 'Board title is empty'
+    });
+  }
+})
+
+exports.getBoard = asyncHandler(async (req, res) => {
+  const { boardId } = req.body;
+  try {
+    const board = await Board.findById(boardId);
+    if (board) {
+      res.status(200).json({
+        board
+      })
+    } else {
+      res.status(404);
+    }
+  } catch (err) {
+    res.status(500);
+    console.error(err);
+  }
+})
+
+exports.createColumn = asyncHandler(async (req, res) => {
+  const { title, boardId } = req.body;
+  if (title) {
```

Comment:
What is there is no boardId passed down? Malicious users who understand enough can edit requests.